### ISSUE TITLE
Should calling `type[T]` with arbitrary arguments be an error?

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#4128]
Should the `cls(1, "foo")` call be considered an error here?
```py
def f[T](cls: type[T]):
    cls()  # all type checker accept this
    cls(1, "foo")  # all type checkers disallow this, except for ty
```

**Comment by AlexWaygood** [comment]
See https://github.com/astral-sh/ruff/pull/23514 for previous detailed discussion
